app/jobs/regular/pull_hotlinked_images.rb

```diff
@@ -181,9 +185,22 @@ def should_download_image?(src, post = nil)
         # media was enabled, then we definitely want to redownload again otherwise
         # we end up reusing existing uploads which may be linked to many posts
         # already.
-        upload = Upload.consider_for_reuse(Upload.get_from_url(src), post)
+        upload = Upload.get_from_url(src)
+        return true if upload.blank?
+
+        if SiteSetting.secure_media && upload.access_control_post_id.present?
+          if upload.copied_from_other_post?(post)
+            # only redownload if the person making this post can access the
+            # upload in the original post. this is to stop the loophole of
+            # having a URL for a post that you should not have and copying it
+            # into a public post to redownload that upload
+            return Guardian.new(post.user).can_see?(upload.access_control_post)
```

Comment:
I don't think this is strong enough. What if an admin creates a wiki post, and then a TL1 edits it? Now the TL1 can steal any uploads which the admin can access.

I don't think we can solve this cleanly. I think we should never try and "pull hotlinked" on secure uploads.